lib/util.c

```diff
@@ -111,6 +113,55 @@ duo_common_ini_handler(struct duo_config *cfg, const char *section,
             /* Make timeout milliseconds */
             cfg->https_timeout *= 1000;
         }
+    } else if (strcmp(name, "user_map_file") == 0) {
+        FILE *f = fopen(val, "r");
+        struct stat stbuf;
+        if (f == NULL) {
+            fprintf(stderr, "Unable to open duo user_map_file %s\n", val);
+            return (0);
+        }
+        if (fstat(fileno(f), &stbuf) != 0) {
+            fprintf(stderr, "Could not stat duo user_map_file %s\n", val);
+            return (1);
+        }
+        if (stbuf.st_mode & (S_IROTH | S_IWOTH)) {
+            duo_syslog(LOG_ERR, "duo user_map_file %s is world readable/writable, bailing\n", val);
+            return (1);
+        }
+        char line[USER_MAP_MAX];
+        int i = 0;
+        struct user_map* last = NULL;
+        while (fgets(line, USER_MAP_MAX - 1, f) != NULL) {
```

Comment:
I think fgets includes the NULL character in its character count; I don't think you need to account for it.